src/jit/emitarm64.cpp

```diff
@@ -10820,13 +10820,13 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
                     if (lsl > 0)
                     {
                         // Generate code to set tmpReg = base + index*scale
-                        emitIns_R_R_R_I(INS_add, EA_PTRSIZE, tmpReg, memBase->gtRegNum, index->gtRegNum, lsl,
+                        emitIns_R_R_R_I(INS_add, EA_BYREF, tmpReg, memBase->gtRegNum, index->gtRegNum, lsl,
```

Comment:
This is making the assumption that the base register is referring to a GC object.  I probably need to check a flag on memBase...

Same below.